### DISCUSSION TITLE
量子機器學習

### DISCUSSION THREAD

**Discussion by EricAgumonJou** [1101-GS3073-C/Python-Community-GS3073-C#285] [General]
量子電腦硬體技術不斷突破，也帶動相關軟體的發展。
前幾年 Google 開發出 「量子機器學習模型的訓練框架」 “TensorFlow Quantum”，讓開發者能藉此開發量子電腦與機器學習應用。

![1*Q5_ub7gT2X5NKgUCJ4MaNQ](https://user-images.githubusercontent.com/91121500/158335928-29f225c5-c58c-407e-bf02-eae7be00a62c.png)
來源：https://towardsdatascience.com/how-may-quantum-computing-benefit-machine-learning-c96de0bef0d4

量子的優點：

- 可跳出 2 位元侷限，量子電腦的運算速度遠大於傳統電腦!!

現在流行的電腦多為傳統電腦，資訊量的基本單位是位元，在二進制中，值“不是 0 就是 1”。

而遵循量子力學規律打造的電腦被稱為量子電腦，資訊量的基本單位是量子位元，在取值前處於不確定狀態，即「疊加態」。
也就是說，量子位元可以同時處於 0 和 1 的狀態。

量子機器學習演算法可分為 3 類：
1. 機器學習的整體計算依然採用傳統電腦，但把比較複雜的計算轉換成量子版本，用量子電腦來運行
2. 尋找量子系統的力學效應、動力學特性與傳統機器學習處理步驟的相似點，將物理過程應用於傳統機器學習問題的求解，產生出新的機器學習演算法。
3. 借助機器學習力去研究量子系統

目前，對量子機器學習的研究大多集中在第一類。在具體的執行層面上，要用量子電腦來運行傳統機器學習複雜計算的部分，首先要把傳統數據轉換成量子數據。

參考資料：https://www.tensorflow.org/quantum/concepts

**Comment by EricAgumonJou** [comment]
雖然量子計算的特性和機器學習相當契合並能達到互補的效果，然而.....整體而言，量子機器學習還處於初級階段。
不是所有的傳統機器學習演算法！！！！，都可以用量子計算加速。

**Comment by EricAgumonJou** [comment]
目前量子機器學習演算法實用化的硬體條件也尚未達到成熟階段。
在通用量子電腦造成功之前，量子機器學習演算法很難應用於實際生活上，以表現其高速數據處理的強大能力。

**Comment by emily40533** [comment]
我對硬體方面、量子等等主題不是很理解，謝謝你發了這篇文來介紹
想請問你文章中那張圖裡，四個象限有甚麼差異呢？尤其是Quantum algorithm type/Classical data type與Classical algorithm type/Quantum data type，謝謝

隨留言附上我對量子的淺薄知識:D
![反正我很閒-量子糾纏](https://user-images.githubusercontent.com/99713899/159417444-be4cbd2d-eeca-480b-b273-849709ffade3.gif)

**Reply by EricAgumonJou** [reply]
關於Classical Data Type和Quantum Data Type比較：
比起傳統電腦位元(bit)只能是0和1，量子電腦的data type（Quantum Data Type）引入物質量子特性，利用其疊加原理進而產生量子位元(qubit)的觀念。

**Reply by EricAgumonJou** [reply]
相對於傳統位元僅有兩種選擇，量子位元有無限多種的選擇。利用此新奇的特性來儲存與操縱信息，可進一步地完成傳統電腦所辦不到的事情—量子霸權(quantum advantage)。

**Reply by EricAgumonJou** [reply]
量子演算法：顧名思義是在量子計算中，於量子計算的現實模型上運行的演算，它最常用的模型是量子線路的計算模型。
傳統演算法：則是有限的指令序列，或用於解決問題的分步驟過程中，每步驟或指令都可在傳統計算機上執行。

**Reply by EricAgumonJou** [reply]
這裏提供一篇關於量子糾纏的文章，供妳參考：）
![6446d860dbbfe540e9e2cbab5f98f1e3-5](https://user-images.githubusercontent.com/91121500/160575153-81f1daa2-5ed7-458a-bf89-1124e7cdf27c.png)

https://pansci.asia/archives/140036
